core/invoker/src/main/scala/whisk/core/invoker/Invoker.scala

```diff
@@ -125,6 +124,14 @@ object Invoker {
 
         new InstanceIdAssigner(config.zookeeperHosts).getId(invokerUniqueName.get)
       }
+
+    // Replace the hostname of the invoker to the assigned id of the invoker.
+    val newKamonConfig = Kamon.config
```

Comment:
Nit can this be refactored to a method `initKamon(hostId:String)`. Further should this be moved to a util method in `whisk.common` such it is also used in Controller which also has this notion of id to identify it.